extensions/vscode/src/utils/errors.ts

```diff
@@ -1,10 +1,28 @@
 // Copyright (C) 2023 by Posit Software, PBC.
 
-import axios from "axios";
+import axios, { AxiosError, AxiosResponse } from "axios";
 
 export type ErrorMessage = string[];
 export type ErrorMessages = ErrorMessage[];
 
+export type ErrorResData = {
+  code: string;
+  status: number;
+  msg: string;
+  error: string;
```

Comment:
Am I correct that you've added an entire structure just so that you can add an error code/string? We already had the code, status and a message within the error response; did you consider simply changing the `error.response.data.msg` from a string to a possible error msg object w/ error string and error string's code?

I think this would have less of an impact on the changes and what you need to do when we have an error string code vs. not. I'm thinking that the agent would change to simply have a new definedError interface which would implement error, plus the new identifying code.

Let me reach out and we can discuss.